lib/trino-hadoop-toolkit/src/main/java/io/trino/hadoop/ConfigurationInstantiator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hadoop;
+
+import org.apache.hadoop.conf.Configuration;
+import org.gaul.modernizer_maven_annotations.SuppressModernizer;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public final class ConfigurationInstantiator
+{
+    private ConfigurationInstantiator() {}
+
+    public static Configuration newEmptyConfiguration()
+    {
+        return newConfiguration(false);
+    }
+
+    /**
+     * @see Configuration#Configuration(boolean)
+     */
+    public static Configuration newConfigurationWithDefaultResources()
+    {
+        return newConfiguration(true);
+    }
+
+    private static Configuration newConfiguration(boolean loadDefaults)
+    {
+        // Configuration captures TCCL and it may used later e.g. to load filesystem implementation class
+        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        ClassLoader expectedClassLoader = ConfigurationInstantiator.class.getClassLoader();
+        checkState(
```

Comment:
This seems silly. If we know exactly which class loader we want, why not just it directly? This is probably what we should have done in the first place rather than the `ClassLoaderSafe` wrappers.

Reply:
I don't think it is silly.

we designed code so that `ClassLoaderSafe*` wrappers handle setting TCCL and imply classloader safety.
doing the check here provides two benefits (1) it provides explicit check in the potentially affected code path, so that we don't silently reintroduce the same problem and (2) remains optional, just a check, so that we can decide to change, refactor or remove this class.